sources/web/datalab/polymer/modules/file-manager-factory/file-manager-factory.ts

```diff
@@ -16,34 +16,54 @@
  * Dependency custom element for ApiManager
  */
 const FILE_MANAGER_ELEMENT = {
+  bigquery: {
+    name: 'bigquery',
+    path: 'modules/bigquery-file-manager/bigquery-file-manager.html',
+    type: BigQueryFileManager,
+  },
   jupyter: {
+    name: 'jupyter',
     path: 'modules/jupyter-file-manager/jupyter-file-manager.html',
     type: JupyterFileManager,
   },
 };
 
+enum FileManagerType {
+  BIG_QUERY,  // bigquery
+  JUPYTER,    // jupyter
+};
+
 /**
  * Maintains and gets the static FileManager singleton.
  */
 // TODO: Find a better way to switch the FileManager instance based on the
 // environment
 class FileManagerFactory {
 
-  private static _fileManager: FileManager;
+  private static _fileManagers: { [fileManagerType: string] : FileManager } = {};
 
+  /** Get the default FileManager. */
   public static getInstance() {
-    if (!FileManagerFactory._fileManager) {
-      const backendType = FileManagerFactory._getBackendType();
+    return FileManagerFactory.getInstanceForType(FileManagerType.JUPYTER);
+  }
+
+  public static getInstanceForType(fileManagerType: FileManagerType) {
+    const backendType = FileManagerFactory._getBackendType(fileManagerType);
+    if (!FileManagerFactory._fileManagers[backendType.name]) {
 
       Polymer.importHref(backendType.path, undefined, undefined, true);
```

Comment:
Let's take this one out for now, and worry about it later if it's causing slow downs. It shouldn't be here if we have the paths imported in the html.